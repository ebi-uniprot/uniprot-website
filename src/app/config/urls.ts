import { invert } from 'lodash-es';

export enum Location {
  Home = 'Home',
  UniProtKBResults = 'UniProtKBResults',
  UniProtKBEntry = 'UniProtKBEntry',
  Blast = 'Blast',
  BlastResult = 'BlastResult',
  UniProtKBCustomiseTable = 'UniProtKBCustomiseTable',
  UniProtKBQueryBuilder = 'UniProtKBQueryBuilder',
  Dashboard = 'Dashboard',
}

export const LocationToPath = {
  [Location.Home]: '/',
  [Location.UniProtKBEntry]: '/uniprotkb/:accession',
  [Location.UniProtKBResults]: '/uniprotkb',
  [Location.BlastResult]: '/blast/:id/:subPage?',
  [Location.Blast]: '/blast',
  [Location.UniProtKBCustomiseTable]: '/customise-table',
  [Location.UniProtKBQueryBuilder]: '/advanced-search',
  [Location.Dashboard]: '/tool-dashboard',
};

export const PathToLocation = invert(LocationToPath);
