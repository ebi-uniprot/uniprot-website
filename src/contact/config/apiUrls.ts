import joinUrl from 'url-join';

import { apiPrefix } from '../../shared/config/apiUrls/apiPrefix';

const token = joinUrl(apiPrefix, 'contact', 'token');

const send = joinUrl(apiPrefix, 'contact', 'send');

export default { token, send };
