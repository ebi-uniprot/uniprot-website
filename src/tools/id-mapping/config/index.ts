export const ID_MAPPING_LIMIT = 100_000;
