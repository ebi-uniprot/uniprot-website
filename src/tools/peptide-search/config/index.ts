export const MAX_PEPTIDE_FACETS_OR_DOWNLOAD = 1_000;
