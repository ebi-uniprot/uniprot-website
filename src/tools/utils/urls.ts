const reIds = /(?<id>\w+)(\[(?<start>\d+)-(?<end>\d+)\])?/;

type IdMaybeWithRange = {
  id: string;
  start?: number;
  end?: number;
};

export const parseIdsFromSearchParams = (
  searchParamIds: string
): IdMaybeWithRange[] =>
  searchParamIds
    .split(',')
    .map((searchParamId): IdMaybeWithRange => {
      const { id, start, end } = searchParamId.match(reIds)?.groups || {};
      return start && end
        ? {
            id,
            start: +start,
            end: +end,
          }
        : { id };
    })
    .filter((idMaybeWithRange): idMaybeWithRange is IdMaybeWithRange =>
      Boolean(idMaybeWithRange)
    );
