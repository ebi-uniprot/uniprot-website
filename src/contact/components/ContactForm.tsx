import { useCallback, FormEventHandler, useRef } from 'react';
import {
  generatePath,
  Link,
  useLocation,
  useRouteMatch,
} from 'react-router-dom';
import {
  Button,
  PageIntro,
  ExternalLink,
  ErrorIcon,
  SuccessIcon,
} from 'franklin-sites';
import { v1 } from 'uuid';
import cn from 'classnames';
import { LocationDescriptor } from 'history';

import HTMLHead from '../../shared/components/HTMLHead';

import { LocationToPath, Location } from '../../app/config/urls';
import postContactForm, {
  ContactFormInputData,
} from '../adapters/contactFormAdapter';

import styles from './styles/contact-form.module.scss';

import HelperContactImage from './svgs/helper-contact.svg';

// ARIA hide all of these, are the state is available in the form already
const validity = (
  <>
    <span aria-hidden="true" className={cn(styles.validity, styles.required)}>
      *
    </span>
    <SuccessIcon
      aria-hidden="true"
      width="1em"
      className={cn(styles.validity, styles.valid)}
    />
    <ErrorIcon
      aria-hidden="true"
      width="1em"
      className={cn(styles.validity, styles.invalid)}
    />
  </>
);

const ContactForm = () => {
  const idRef = useRef(v1());
  const isUpdate = !!useRouteMatch(LocationToPath[Location.ContactUpdate]);
  const referrer =
    useLocation<{ referrer?: LocationDescriptor }>().state?.referrer;

  console.log('referrer', referrer);

  const privacyCheckBox = useCallback((checkbox: HTMLInputElement) => {
    checkbox.setCustomValidity('Please tick the box to agree.');
  }, []);

  const handleSubmit = useCallback<FormEventHandler<HTMLFormElement>>(
    (event) => {
      const form = event.target;
      event.preventDefault();
      if (!(form instanceof HTMLFormElement)) {
        return;
      }
      // Parse form and generate form input data
      const contactFormInputData: ContactFormInputData = {
        email: (form.elements.namedItem('email') as HTMLInputElement).value,
        name: (form.elements.namedItem('name') as HTMLInputElement).value,
        subject: (form.elements.namedItem('subject') as HTMLInputElement).value,
        message: (form.elements.namedItem('message') as HTMLInputElement).value,
      };
      postContactForm(contactFormInputData);
      event.preventDefault();
    },
    []
  );

  const description = isUpdate
    ? 'Submit updates or corrections to UniProt'
    : 'Send us general questions and suggestions using the form below';

  const id = idRef.current;

  return (
    <>
      <HTMLHead title="Contact us" />
      <PageIntro title="Contact us" />
      <section className={styles.container}>
        <h2 className="medium">{description}</h2>
        <hr />
        <form aria-label="Contact form" onSubmit={handleSubmit}>
          {/* Name */}
          <label className={styles.label} htmlFor={`name-${id}`}>
            Name:
          </label>
          <span className={styles.input}>
            <input
              type="text"
              name="name"
              placeholder=" "
              id={`name-${id}`}
              maxLength={100}
            />
            {validity}
          </span>
          {/* E-mail */}
          <label className={styles.label} htmlFor={`email-${id}`}>
            E-mail:
          </label>
          <span className={styles.input}>
            <input
              type="email"
              name="email"
              placeholder="myemail@example.com"
              id={`email-${id}`}
              required
              minLength={4}
              maxLength={100}
            />
            {validity}
          </span>
          {/* Subject */}
          <label className={styles.label} htmlFor={`subject-${id}`}>
            Subject:
          </label>
          <span className={styles.input}>
            <input
              type="text"
              name="subject"
              placeholder=" "
              id={`subject-${id}`}
              required
              minLength={1}
              maxLength={100}
            />
            {validity}
          </span>
          {/* Message */}
          <label className={styles.label} htmlFor={`message-${id}`}>
            Message:
          </label>
          <span className={cn(styles.input, styles.input__message)}>
            <textarea
              name="message"
              placeholder="my message"
              id={`message-${id}`}
              required
              minLength={1}
            />
            {validity}
          </span>
          {/* Privacy */}
          <label className={styles.privacy}>
            <input
              type="checkbox"
              name="privacy"
              required
              ref={privacyCheckBox}
            />
            I agree to the processing of my data for the purposes described in
            this{' '}
            <Link
              to={generatePath(LocationToPath[Location.HelpEntry], {
                accession: 'privacy',
              })}
            >
              privacy notice
            </Link>
            .
          </label>
          {/* üçØ */}
          <input
            type="text"
            name="requiredForRobots"
            // Make sure it's NOT reachable, on purpose
            tabIndex={-1}
            aria-hidden="true"
          />
          <Button type="submit">Send message</Button>
          {/* TODO: after TRM-25295, make sure this is not even loaded on
           * smaller screens when loaded as an image */}
          <HelperContactImage
            className={styles.illustration}
            // override the viewBox in order to make it disappear a bit
            viewBox="0 0 211 140"
          />
          <aside>
            <h2 className="small">Other ways to contact us</h2>
            {/* Link to switch the contact form */}
            <div>
              <Link
                to={{
                  pathname:
                    LocationToPath[
                      isUpdate
                        ? Location.ContactGeneric
                        : Location.ContactUpdate
                    ],
                  // Make sure to pass along the previour referrer if switching
                  state: { referrer },
                }}
              >
                {isUpdate
                  ? 'Send us general questions and suggestions'
                  : 'Send updates or corrections'}
              </Link>
            </div>
            <div>
              <ExternalLink url="https://www.ebi.ac.uk/swissprot/Submissions/spin/">
                Submit new protein sequence data
              </ExternalLink>
            </div>
          </aside>
        </form>
      </section>
    </>
  );
};

export default ContactForm;
