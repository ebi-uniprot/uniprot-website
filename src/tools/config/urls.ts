import queryString from 'query-string';

import { createFacetsQueryString } from '../../shared/config/apiUrls';
import { SelectedFacet } from '../../uniprotkb/types/resultsTypes';
import { JobTypes } from '../types/toolsJobTypes';

type CommonResultFormats =
  | 'out' // raw output of the tool
  | 'sequence'; // raw, as submitted

export type ResultFormat = {
  [JobTypes.ALIGN]:
    | CommonResultFormats
    // Alignment in CLUSTAL format with base/residue numbering;
    // https://www.ebi.ac.uk/seqdb/confluence/display/JDSAT/Multiple+Sequence+Alignment+Tool+Output+Examples#MultipleSequenceAlignmentToolOutputExamples-ClustalOmegaproteinoutputexamples
    // ClustalW:
    // http://web.mit.edu/meme_v4.11.4/share/doc/clustalw-format.html
    | 'aln-clustal_num'
    | 'phylotree' // Phylogenetic Tree
    | 'pim' // Percent Identity Matrix
    | 'submission'; // Submission Details (according to doc in JSON, but it looks like it's actually XML)
  [JobTypes.BLAST]:
    | CommonResultFormats
    | 'ids' // e.g. 'TR:F8D2X3_HALXS'
    | 'accs' // e.g. 'TR:F8D2X3'
    | 'xml'
    | 'visual-svg'
    | 'complete-visual-svg'
    | 'visual-png'
    | 'complete-visual-png'
    | 'visual-jpg'
    | 'complete-visual-jpg'
    | 'ffdp-query-svg'
    | 'ffdp-query-png'
    | 'ffdp-query-jpeg'
    | 'ffdp-subject-svg'
    | 'ffdp-subject-png'
    | 'ffdp-subject-jpeg'
    | 'parameters' // in XML
    | 'json';
  [JobTypes.ID_MAPPING]: never;
  [JobTypes.PEPTIDE_SEARCH]: never;
};

type Return<T extends JobTypes> = Readonly<{
  runUrl: string;
  statusUrl: (jobId: string) => string;
  resultUrl: (
    redirectUrl: string,
    extra: {
      format?: ResultFormat[T];
      facets?: string[];
      size?: number;
      selectedFacets?: SelectedFacet[];
    }
  ) => string;
  detailsUrl?: (jobId: string) => string;
}>;

function urlObjectCreator<T extends JobTypes>(type: T): Return<T> {
  let baseURL = '';
  switch (type) {
    case JobTypes.ALIGN:
      baseURL = 'https://www.ebi.ac.uk/Tools/services/rest/clustalo';
      break;
    case JobTypes.BLAST:
      baseURL = 'https://www.ebi.ac.uk/Tools/services/rest/ncbiblast';
      break;
    case JobTypes.ID_MAPPING:
      baseURL = 'https://wwwdev.ebi.ac.uk/uniprot/api/idmapping';
      return Object.freeze({
        runUrl: `${baseURL}/run`,
        statusUrl: (jobId) =>
          // The cachebust extra query is just here to avoid using cached value
          `${baseURL}/status/${jobId}?cachebust=${new Date().getTime()}`,
        resultUrl: (redirectUrl, { facets, size, selectedFacets = [] }) =>
          `https://wwwdev.ebi.ac.uk${redirectUrl}?${queryString.stringify({
            size,
            facets: facets?.join(','),
            query: createFacetsQueryString(selectedFacets),
          })}`,
        detailsUrl: (jobId) => `${baseURL}/details/${jobId}`,
      });
      break;
    case JobTypes.PEPTIDE_SEARCH:
      baseURL =
        'https://research.bioinformatics.udel.edu/peptidematchws/asyncrest';
      return Object.freeze({
        runUrl: baseURL,
        statusUrl: (jobId) => `${baseURL}/jobs/${jobId}`,
        resultUrl: (jobId) => `${baseURL}/jobs/${jobId}`,
      });
    default:
    //
  }
  return Object.freeze({
    runUrl: `${baseURL}/run`,
    statusUrl: (jobId) => `${baseURL}/status/${jobId}`,
    resultUrl: (jobId, { format }) => `${baseURL}/result/${jobId}/${format}`,
  });
}

export default urlObjectCreator;
