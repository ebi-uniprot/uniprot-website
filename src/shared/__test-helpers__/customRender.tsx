/**
 * This tries to follow what is detailed here:
 * https://testing-library.com/docs/react-testing-library/setup#custom-render
 */
import { render, RenderOptions } from '@testing-library/react';
import { Component, Dispatch, ReactElement, ReactNode } from 'react';
import { HelmetProvider } from 'react-helmet-async';
import {
  createMemoryHistory,
  LocationState,
  MemoryHistory,
  Route,
  Router,
} from 'react-router';
import { JsonValue, SetRequired } from 'type-fest';

import { MessagesState } from '../../messages/state/messagesInitialState';
import { MessagesAction } from '../../messages/state/messagesReducers';
import databaseInfo from '../../uniprotkb/utils/__tests__/__mocks__/databaseInfo';
import { getDatabaseInfoMaps } from '../../uniprotkb/utils/database';
import {
  MessagesDispatchContext,
  MessagesStateContext,
} from '../contexts/Messages';
import {
  DatabaseInfoMapsContext,
  UniProtDataVersionContext,
} from '../contexts/UniProtData';
import { JobAction } from '../workers/jobs/state/actionHandler';
import { JobsState } from '../workers/jobs/state/jobsInitialState';

type ExtraRenderOptions = {
  // For react-router
  history?: MemoryHistory<LocationState>;
  path?: string;
  // For custom user preferences (used by useLocalStorage)
  initialLocalStorage?: Record<string, JsonValue>;
  // For app context
  messagesState?: MessagesState;
  messagesDispatch?: jest.Mock<Dispatch<MessagesAction>>;
  JobsState?: JobsState;
  toolsDispatch?: jest.Mock<Dispatch<JobAction>>;
};

type WrapperProps = { children: ReactNode } & RenderOptions &
  SetRequired<
    ExtraRenderOptions,
    | 'history'
    | 'initialLocalStorage'
    | 'messagesState'
    | 'messagesDispatch'
    | 'JobsState'
    | 'toolsDispatch'
  >;

const dbInfoMaps = getDatabaseInfoMaps(databaseInfo);
const uniProtDataVersion = {
  releaseNumber: '2023_01',
  releaseDate: new Date('01-March-2023'),
};

class Wrapper extends Component<WrapperProps> {
  constructor(props: WrapperProps) {
    super(props);
    for (const [key, value] of Object.entries(props.initialLocalStorage)) {
      window.localStorage.setItem(key, JSON.stringify(value));
    }
  }

  // eslint-disable-next-line class-methods-use-this
  componentWillUnmount() {
    window.localStorage.clear();
  }

  render() {
    const { children, path, history, messagesDispatch, messagesState } =
      this.props;
    return (
      <HelmetProvider>
        <Router history={history}>
          <MessagesDispatchContext.Provider value={messagesDispatch}>
            <MessagesStateContext.Provider value={messagesState}>
              <DatabaseInfoMapsContext.Provider value={dbInfoMaps}>
                <UniProtDataVersionContext.Provider value={uniProtDataVersion}>
                  {path ? (
                    <Route path={path} render={() => children} />
                  ) : (
                    children
                  )}
                </UniProtDataVersionContext.Provider>
              </DatabaseInfoMapsContext.Provider>
            </MessagesStateContext.Provider>
          </MessagesDispatchContext.Provider>
        </Router>
      </HelmetProvider>
    );
  }
}

const customRender = (
  ui: ReactElement,
  {
    route = '',
    path,
    history = createMemoryHistory({ initialEntries: [route] }),
    initialLocalStorage = {},
    messagesState = {},
    messagesDispatch = jest.fn(),
    JobsState = {},
    toolsDispatch = jest.fn(),
    ...options
  }: RenderOptions &
    ExtraRenderOptions & {
      route?: string;
    } = {}
) => ({
  ...render(ui, {
    wrapper: (props) => (
      <Wrapper
        path={path}
        history={history}
        initialLocalStorage={initialLocalStorage}
        messagesState={messagesState}
        messagesDispatch={messagesDispatch}
        JobsState={JobsState}
        toolsDispatch={toolsDispatch}
        {...props}
      />
    ),
    ...options,
  }),
  history,
  messagesDispatch,
  toolsDispatch,
});

export default customRender;
