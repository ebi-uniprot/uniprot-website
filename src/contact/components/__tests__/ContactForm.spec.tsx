import { screen } from '@testing-library/react';

import customRender from '../../../shared/__test-helpers__/customRender';

import ContactForm from '../ContactForm';

import {
  useFormLogic,
  UseFormLogicReturnType,
} from '../../adapters/contactFormAdapter';

import { Location, LocationToPath } from '../../../app/config/urls';

jest.mock('../../adapters/contactFormAdapter');
const handleSubmit = jest.fn();

describe('ContactForm', () => {
  beforeEach(async () => {
    handleSubmit.mockClear();
    (useFormLogic as jest.Mock<UseFormLogicReturnType>).mockReturnValue({
      sending: false,
      handleSubmit,
    });
  });

  it('should render for generic contact form', () => {
    const { asFragment } = customRender(<ContactForm />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render for feedback contact form', () => {
    const { asFragment } = customRender(<ContactForm />, {
      route: LocationToPath[Location.ContactUpdate],
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should prefill the form for feedback contact form', () => {
    customRender(<ContactForm />, {
      route: `${
        LocationToPath[Location.ContactUpdate]
      }?entry=P05067&entryType=Reviewed (Swiss-Prot)`,
    });
    expect(screen.getByRole('textbox', { name: /subject/i })).toHaveValue(
      'Reviewed (Swiss-Prot) P05067 entry update request'
    );
  });

  it('should be disabled when submitting', () => {
    (useFormLogic as jest.Mock<UseFormLogicReturnType>).mockReturnValue({
      sending: true,
      handleSubmit,
    });
    const { asFragment } = customRender(<ContactForm />);
    expect(asFragment()).toMatchSnapshot();
    const button = screen.getByRole<HTMLButtonElement>('button');
    expect(button).toBeDisabled();
  });
});
