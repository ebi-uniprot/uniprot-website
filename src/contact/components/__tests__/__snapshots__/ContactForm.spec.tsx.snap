// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should be disabled when submitting 1`] = `
<DocumentFragment>
  <div
    class="page-intro"
  >
    <h1
      class=""
    >
      Contact us
    </h1>
  </div>
  <section
    class="container"
  >
    <h2
      class="medium"
    >
      Send us general questions and suggestions using the form below
    </h2>
    <hr />
    <form
      aria-label="Contact form"
    >
      <label
        class="label"
        for="name-0"
      >
        Name:
      </label>
      <span
        class="input"
      >
        <input
          id="name-0"
          maxlength="100"
          name="name"
          placeholder=" "
          type="text"
          value=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="email-0"
      >
        E-mail:
      </label>
      <span
        class="input"
      >
        <input
          id="email-0"
          maxlength="100"
          minlength="4"
          name="email"
          placeholder="myemail@example.com"
          required=""
          type="email"
          value=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="subject-0"
      >
        Subject:
      </label>
      <span
        class="input"
      >
        <input
          id="subject-0"
          maxlength="100"
          minlength="1"
          name="subject"
          placeholder=" "
          required=""
          type="text"
          value=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="message-0"
      >
        Message:
      </label>
      <span
        class="input input__message"
      >
        <textarea
          id="message-0"
          minlength="1"
          name="message"
          placeholder="my message"
          required=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="privacy"
      >
        <input
          name="privacy"
          required=""
          type="checkbox"
        />
        I agree to the processing of my data for the purposes described in this 
        <a
          href="/help/privacy"
        >
          privacy notice
        </a>
        .
        <span
          aria-hidden="true"
          class="required"
        >
           *
        </span>
      </label>
      <input
        aria-hidden="true"
        name="requiredForRobots"
        tabindex="-1"
        type="text"
      />
      <input
        hidden=""
        name="referrer"
        readonly=""
        value=""
      />
      <button
        class="button primary disabled"
        disabled=""
        type="submit"
      >
        Sending message
      </button>
      <test-file-stub
        classname="illustration"
        viewbox="0 0 211 140"
      />
      <aside>
        <h2
          class="small"
        >
          Other ways to contact us
        </h2>
        <ul
          class="no-bullet"
        >
          <li>
            <a
              href="/update"
            >
              Send updates or corrections
            </a>
          </li>
          <li>
            <a
              class="external-link"
              href="https://www.ebi.ac.uk/swissprot/Submissions/spin/"
              rel="noopener"
              target="_blank"
            >
              Submit new protein sequence data
              <test-file-stub
                data-testid="external-link-icon"
                width="12.5"
              />
            </a>
          </li>
        </ul>
      </aside>
    </form>
    <div
      class="uniprot-grid-cell--span-12 message message--info"
      role="status"
    >
      <div
        class="message__side-border"
      />
      <test-file-stub
        height="1.125em"
        width="1.125em"
      />
      <section
        class="message__content"
      >
        <small>
          If you still need it, the 
          <a
            class="external-link"
            href="https://legacy.uniprot.org"
            rel="noopener nofollow"
            target="_blank"
          >
            legacy version of the website
          </a>
           is available until the 2022_03 release.
        </small>
      </section>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ContactForm should render for feedback contact form 1`] = `
<DocumentFragment>
  <div
    class="page-intro"
  >
    <h1
      class=""
    >
      Contact us
    </h1>
  </div>
  <section
    class="container"
  >
    <h2
      class="medium"
    >
      Submit updates or corrections to UniProt
    </h2>
    <hr />
    <form
      aria-label="Contact form"
    >
      <label
        class="label"
        for="name-0"
      >
        Name:
      </label>
      <span
        class="input"
      >
        <input
          id="name-0"
          maxlength="100"
          name="name"
          placeholder=" "
          type="text"
          value=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="email-0"
      >
        E-mail:
      </label>
      <span
        class="input"
      >
        <input
          id="email-0"
          maxlength="100"
          minlength="4"
          name="email"
          placeholder="myemail@example.com"
          required=""
          type="email"
          value=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="subject-0"
      >
        Subject:
      </label>
      <span
        class="input"
      >
        <input
          id="subject-0"
          maxlength="100"
          minlength="1"
          name="subject"
          placeholder=" "
          required=""
          type="text"
          value="UniProtKB entry update request"
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="message-0"
      >
        Message:
      </label>
      <span
        class="input input__message"
      >
        <textarea
          id="message-0"
          minlength="1"
          name="message"
          placeholder="my message"
          required=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="privacy"
      >
        <input
          name="privacy"
          required=""
          type="checkbox"
        />
        I agree to the processing of my data for the purposes described in this 
        <a
          href="/help/privacy"
        >
          privacy notice
        </a>
        .
        <span
          aria-hidden="true"
          class="required"
        >
           *
        </span>
      </label>
      <input
        aria-hidden="true"
        name="requiredForRobots"
        tabindex="-1"
        type="text"
      />
      <input
        hidden=""
        name="referrer"
        readonly=""
        value=""
      />
      <button
        class="button primary"
        type="submit"
      >
        Send message
      </button>
      <test-file-stub
        classname="illustration"
        viewbox="0 0 211 140"
      />
      <aside>
        <h2
          class="small"
        >
          Other ways to contact us
        </h2>
        <ul
          class="no-bullet"
        >
          <li>
            <a
              href="/contact"
            >
              Send us general questions and suggestions
            </a>
          </li>
          <li>
            <a
              class="external-link"
              href="https://www.ebi.ac.uk/swissprot/Submissions/spin/"
              rel="noopener"
              target="_blank"
            >
              Submit new protein sequence data
              <test-file-stub
                data-testid="external-link-icon"
                width="12.5"
              />
            </a>
          </li>
        </ul>
      </aside>
    </form>
    <div
      class="uniprot-grid-cell--span-12 message message--info"
      role="status"
    >
      <div
        class="message__side-border"
      />
      <test-file-stub
        height="1.125em"
        width="1.125em"
      />
      <section
        class="message__content"
      >
        <small>
          If you still need it, the 
          <a
            class="external-link"
            href="https://legacy.uniprot.org"
            rel="noopener nofollow"
            target="_blank"
          >
            legacy version of the website
          </a>
           is available until the 2022_03 release.
        </small>
      </section>
    </div>
  </section>
</DocumentFragment>
`;

exports[`ContactForm should render for generic contact form 1`] = `
<DocumentFragment>
  <div
    class="page-intro"
  >
    <h1
      class=""
    >
      Contact us
    </h1>
  </div>
  <section
    class="container"
  >
    <h2
      class="medium"
    >
      Send us general questions and suggestions using the form below
    </h2>
    <hr />
    <form
      aria-label="Contact form"
    >
      <label
        class="label"
        for="name-0"
      >
        Name:
      </label>
      <span
        class="input"
      >
        <input
          id="name-0"
          maxlength="100"
          name="name"
          placeholder=" "
          type="text"
          value=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="email-0"
      >
        E-mail:
      </label>
      <span
        class="input"
      >
        <input
          id="email-0"
          maxlength="100"
          minlength="4"
          name="email"
          placeholder="myemail@example.com"
          required=""
          type="email"
          value=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="subject-0"
      >
        Subject:
      </label>
      <span
        class="input"
      >
        <input
          id="subject-0"
          maxlength="100"
          minlength="1"
          name="subject"
          placeholder=" "
          required=""
          type="text"
          value=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="label"
        for="message-0"
      >
        Message:
      </label>
      <span
        class="input input__message"
      >
        <textarea
          id="message-0"
          minlength="1"
          name="message"
          placeholder="my message"
          required=""
        />
        <span
          aria-hidden="true"
          class="validity required"
        >
          *
        </span>
        <test-file-stub
          aria-hidden="true"
          classname="validity valid"
          width="1em"
        />
        <test-file-stub
          aria-hidden="true"
          classname="validity invalid"
          width="1em"
        />
      </span>
      <label
        class="privacy"
      >
        <input
          name="privacy"
          required=""
          type="checkbox"
        />
        I agree to the processing of my data for the purposes described in this 
        <a
          href="/help/privacy"
        >
          privacy notice
        </a>
        .
        <span
          aria-hidden="true"
          class="required"
        >
           *
        </span>
      </label>
      <input
        aria-hidden="true"
        name="requiredForRobots"
        tabindex="-1"
        type="text"
      />
      <input
        hidden=""
        name="referrer"
        readonly=""
        value=""
      />
      <button
        class="button primary"
        type="submit"
      >
        Send message
      </button>
      <test-file-stub
        classname="illustration"
        viewbox="0 0 211 140"
      />
      <aside>
        <h2
          class="small"
        >
          Other ways to contact us
        </h2>
        <ul
          class="no-bullet"
        >
          <li>
            <a
              href="/update"
            >
              Send updates or corrections
            </a>
          </li>
          <li>
            <a
              class="external-link"
              href="https://www.ebi.ac.uk/swissprot/Submissions/spin/"
              rel="noopener"
              target="_blank"
            >
              Submit new protein sequence data
              <test-file-stub
                data-testid="external-link-icon"
                width="12.5"
              />
            </a>
          </li>
        </ul>
      </aside>
    </form>
    <div
      class="uniprot-grid-cell--span-12 message message--info"
      role="status"
    >
      <div
        class="message__side-border"
      />
      <test-file-stub
        height="1.125em"
        width="1.125em"
      />
      <section
        class="message__content"
      >
        <small>
          If you still need it, the 
          <a
            class="external-link"
            href="https://legacy.uniprot.org"
            rel="noopener nofollow"
            target="_blank"
          >
            legacy version of the website
          </a>
           is available until the 2022_03 release.
        </small>
      </section>
    </div>
  </section>
</DocumentFragment>
`;
